### DISCUSSION TITLE
Future of the gitter channel

### DISCUSSION THREAD

**Discussion by weiznich** [diesel-rs/diesel#4266] [Polls]
I'm lately more and more annoyed with the current state of the gitter/matrix channel mostly due to how the Matrix is not working. In detail

* I somehow got stripped of the admin rights of the channel again
* There seems to be more and more spam
* The mobile clients are not really functional for me
* Session state is lost for me for each new login in the same browser

Given that and that I have the feeling that the channel is much less active than in the past I think about closing it down. That raises the question: Do we need another chat solution or should we just shut down the channel without replacement? If we want to have a replacement what would be a good solution?

**Comment by dessalines** [comment]
I'm guessing this is because of the inadaquate way gitter integrates with matrix? Might be better to just create a fresh matrix room w/o gitter.

**Reply by weiznich** [reply]
That would at least address my first point, but I fear that it wouldn't change the last three points as those are more fundamental limitations of Matrix?

**Reply by dessalines** [reply]
We run some larger open matrix rooms for lemmy, and the spam isn't too extreme, maybe once or twice a month, and they can be banned / removed easily. That's a problem you're going to have no matter the chat platform. You can also keep the room unpublished, but available to join by link, which might cut down on spam (but most projects don't do that).

There are a lot of well-done mobile clients now for matrix, especially element, but there's also schildichat and a few others. Not perfect of course, but the app ecosystem is bigger than almost anything else you'd find.

I've never had that problem with session state, and I don't remember it being a problem with element-web (maybe that's another gitter related problem). I've def never had it with element-android.

**Reply by weiznich** [reply]
I'm already using element-web and element for ios. Maybe I'm just unlucky or it's the room, but I often end up in situations where it tells me that there are new notifications and then there is nothing. 

As with the session state: That might actually be my browser configuration, but it seems to lose the verification state every time I restart the browser. 

The spam problem is mostly problematic in combination with the lose of admin rights, otherwise I would be able to handle it on my own.

**Comment by linrongbin16** [comment]
Some projects are using discord, which has a great App for both web and mobile.

**Reply by weiznich** [reply]
I'm personally not a big fan of discords user interface as I do not find it well organized for complex conversations that might span more than a day. I my experience it's really easy to loose track of previous parts of the same discussion.

Also discord is not an open platform, especially as for as I know it's not possible to access information from the chats without logins which is something that is a huge downside for a open source project in my opinion.

**Reply by linrongbin16** [reply]
for long conversations, github issue/discussion is the best.

chatting room is just for chatting.

**Reply by weiznich** [reply]
My experience with having this gitter room for quite some while indicates that quite a few people use these platforms differently. Especially common are cases where people are in opposite time zones, which quickly results in conversions spanning multiple days just due to the time differences.

**Reply by YievCkim** [reply]
> chatting room is just for chatting.

I am agree but I also found that responsiveness was superior on chat.

Not a fan of discord too.

**Comment by DasNaCl** [comment]
Looks like, as of now, most people voted for "Something else (please comment below)" without commenting.
I'd be intrigued to find a better alternative than zulip. I think this would be the way to go and am happy that you consider it, weiznich!

**Reply by weiznich** [reply]
As for as I can see 3 out of those 4 votes are represented by a comment. One seems to be [that comment](https://github.com/diesel-rs/diesel/discussions/4266#discussioncomment-10666441), while the other two seems to be [this comment](https://github.com/diesel-rs/diesel/discussions/4266#discussioncomment-10665461) (2 upvotes)

**Comment by Sytten** [comment]
Gitter is a mess. I would say let's just use github discussions. They are indexed and we can increase the diesel knowledge base that way.

**Comment by Sagebati** [comment]
Personally, I liked matrix because having others real time chats for other libraries (Rocket, Axum, Rust...) in the same place was pleasant.

If it's shutdown, Github Discussions seems nice. But doesn't fill the same functionality.

**Comment by jblachly** [comment]
Vote for: start a new room with higher version number (room is v1; current matrix rooms are v10 or 11); will require admin privs (I DM'ed you about this) to keep the same name.

If staying with matrix (same name or new name), ditch the GItter integration

**Reply by peter-scholtens** [reply]
1. I agree. Let's have more people with the admin role (let's say 2 or 3), as it is a way too common problem that an admin loses login credentials by hardware failure or the [bus factor](https://en.wikipedia.org/wiki/Bus_factor). I don't mind kicking or banning spammers as long as several  people have this access right too, so quick support for this does NOT rest on the shoulders of __only one person__.
2. Migration to a new Matrix channel may work, but we may loose some of the > 3000 current participants in the chat, even with a re-direction link (But not all of them may be very active anymore). Removing the old channel afterwards could be done by a request to matrix.org admins.
3. I prefer to pay _and_ have influence via e.g. voting, at a community supported sites like [Codeberg](https://join.codeberg.org/) instead of paying with my privacy or information being sold like this site (they have no independent Matrix server yet).
4. I guess we can ask [them](https://matrix.org/contact/) so elevate some of us (the current Matrix users) to the admin level?

**Reply by weiznich** [reply]
@peter-scholtens 

Just to clarify a few points here: The original Gitter/matrix channel had more than one admin. Somehow the got removed recently for whatever reason without any notification to the removed users. jblachly also already tried to reach out to the matrix folks about that more than a month ago, but nothing has happened from their side so far. Given that I do not really see other options to abandon that channel and given that experience I'm not sure I would trust Matrix again to host another support channel. At that point I really do not see other reasonable options than sticking to GitHub for now. Sure it would be nice to have an open solution there, but I cannot see one that reliably works and already have the necessary user base to be reasonably useful. Codeberg is unfortunately not there yet, hopefully that changes at some point in the future.

**Comment by marvin-hansen** [comment]
I vote for just use Github discussions / issues if this is still relevant.

I found so much hidden knowledge in GH issues and discussions just by Googling around, which saved me a ton
of time asking around and, when I asked, I could at least point at some stuff  in the repo that helped to elaborate the issue at hand.

**Comment by YievCkim** [comment]
I voted for zulip but another room on matrix is ok for me. 

I'd like to take this opportunity to point out that the site still links to the gitter and it might be interesting to point this out or remove the link altogether.

**Reply by weiznich** [reply]
> I'd like to take this opportunity to point out that the site still links to the gitter and it might be interesting to point this out or remove the link altogether.

Could you point to the exact location on the webpage that still references the Gitter channel? That's something we want to remove from there.

**Reply by YievCkim** [reply]
I think I found the link from the README on GH https://github.com/diesel-rs/diesel?tab=readme-ov-file#getting-help

(Sorry I didn't remember I saw this link on the README and not on the website. )

**Reply by weiznich** [reply]
No need to be sorry. It's just that I removed some links to Gitter in the last week and I wondered where I missed one, so thanks for pointing to the right location. I will remove that one as well.
